openshift-kube-apiserver/filters/deprecatedapirequest/update_func.go

```diff
@@ -0,0 +1,111 @@
+package deprecatedapirequest
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apiv1 "github.com/openshift/api/apiserver/v1"
+	"k8s.io/kubernetes/openshift-kube-apiserver/filters/deprecatedapirequest/v1helpers"
+)
+
+// IncrementRequestCounts add additional api request counts to the log.
+func SetRequestCountsForNode(node string, requestCounts *resourceRequestCounts) v1helpers.UpdateStatusFunc {
+	return func(status *apiv1.DeprecatedAPIRequestStatus) {
+		currentHour := time.Now().Hour()
+		for hour, hourlyCounts := range requestCounts.hours {
+			lastUpdateTime := hourlyCounts.lastUpdateTime
+			for len(status.RequestsLast24h) <= hour {
+				status.RequestsLast24h = append(status.RequestsLast24h, apiv1.RequestLog{})
+			}
+			for user, users := range hourlyCounts.users {
+				for verb, verbs := range users.verbs {
+					status.RequestsLast24h[hour] = setRequestCount(status.RequestsLast24h[hour], node, lastUpdateTime, user, verb, int(verbs.count))
+				}
+			}
+
+			for len(status.RequestsLast24h) <= hour {
+				status.RequestsLast24h = append(status.RequestsLast24h, apiv1.RequestLog{})
+			}
+		}
+		for len(status.RequestsLast24h) <= currentHour {
+			status.RequestsLast24h = append(status.RequestsLast24h, apiv1.RequestLog{})
+		}
+		status.RequestsLastHour = status.RequestsLast24h[currentHour]
+	}
+}
+
+func setRequestCount(log apiv1.RequestLog, nodeName string, lastUpdateTime time.Time, verb, username string, count int) apiv1.RequestLog {
```

Comment:
this function isn't working for you. it's hard to read and it's not obvious how you're sorting and limiting the total number of  users.